lib/orderbook/TradingPair.ts

```diff
@@ -100,47 +105,36 @@ class MatchingEngine {
    * @returns false if it's a duplicated order, otherwise true
    */
   public addPeerOrder = (order: StampedPeerOrder): boolean => {
-    let peerOrdersList = this.peersOrders.get(order.peerPubKey);
-    if (!peerOrdersList) {
-      peerOrdersList = {
+    let peerOrdersMaps = this.peersOrders.get(order.peerPubKey);
+    if (!peerOrdersMaps) {
+      peerOrdersMaps = {
         buy: new Map<string, StampedPeerOrder>(),
         sell: new Map<string, StampedPeerOrder>(),
       };
-      this.peersOrders.set(order.peerPubKey, peerOrdersList);
+      this.peersOrders.set(order.peerPubKey, peerOrdersMaps);
     }
-    return this.addOrder(order, peerOrdersList);
-  }
 
-  // TODO: remove method
-  /**
-   * Match an order against its opposite queue, and optionally add the unmatched portion of the order to the queue.
-   * @param discardRemaining whether to discard any unmatched portion of the order rather than add it to the queue
-   * @returns a [[MatchingResult]] with the matches as well as the remaining, unmatched portion of the order
-   */
-  public matchOrAddOwnOrder = (order: StampedOwnOrder, discardRemaining: boolean): matchingEngine.MatchingResult => {
-    const matchingResult = this.match(order);
-    if (matchingResult.remainingOrder && !discardRemaining) {
-      this.addOwnOrder(matchingResult.remainingOrder);
-    }
-
-    return matchingResult;
+    return this.addOrder(order, peerOrdersMaps);
   }
 
   public addOwnOrder = (order: StampedOwnOrder): boolean => {
     return this.addOrder(order, this.ownOrders);
   }
 
-  private addOrder = (order: StampedOrder, lists: OrderSidesLists<StampedOrder>): boolean => {
-    const list = order.isBuy ? lists.buy : lists.sell;
-    const queue = order.isBuy ? this.queues.buy : this.queues.sell;
-
-    if (list.has(order.id)) {
+  private addOrder = (order: StampedOrder, maps: OrderSidesMaps<StampedOrder>): boolean => {
+    const map = order.isBuy ? maps.buy : maps.sell;
+    if (map.has(order.id)) {
       return false;
```

Comment:
If we'll throw an error, we need to be careful with a peerOrder case. At the moment we're having the relaxed `boolean` result, and it's applied on `OrderBook.addOwnOrder` with an assert, because it's an unexpected behavior (id supposed to be unique).

I'm fine with the current behavior, let me know if you think otherwise.

Reply:
OK, yeah I'll maybe take another look at this behavior later but it's fine for now.